datafusion/functions-aggregate/src/count.rs

```diff
@@ -139,6 +148,185 @@ impl AggregateUDFImpl for Count {
         "count"
     }
 
+    fn schema_name(&self, params: &AggregateFunctionParams) -> Result<String> {
+        let AggregateFunctionParams {
+            args,
+            distinct,
+            filter,
+            order_by,
+            null_treatment,
+        } = params;
+
+        let mut schema_name = String::new();
+
+        if !args.is_empty() && args[0] == Expr::Literal(COUNT_STAR_EXPANSION) {
+            schema_name.write_str("count(*)")?;
```

Comment:
- `count(1)` and `count(2)` are the same thing, so what about checking for args[0] to be a non-null constant?

- `count(1, a, b)` is something else than `count(1)`; this should check that args.len = 1